test/ngAnimate/integrationSpec.js

```diff
@@ -499,6 +571,45 @@ describe('ngAnimate integration tests', function() {
         expect(child).not.toHaveClass('blue');
       });
     });
+
+    it('should cache invalid animations to avoid application of ngAnimate CSS classes', function() {
+      function fill(max) {
+        var arr = [];
+        for (var i = 0; i < max; i++) {
+          arr.push(i);
+        }
+        return arr;
+      }
+
+      inject(function($animate, $rootScope, $compile, $timeout, $$rAF, $$jqLite) {
+        ss.addRule('.animate-me', 'transition: all 0.5s;');
+
+        var classAddSpy = spyOn($$jqLite, 'addClass').andCallThrough();
+        var classRemoveSpy = spyOn($$jqLite, 'removeClass').andCallThrough();
+
+        element = jqLite(
+          '<div>' +
+            '<div ng-repeat="item in items"></div>' +
+          '</div> '
+        );
+
+        html(element);
+        $compile(element)($rootScope);
+
+        $rootScope.items = fill(100);
+        $rootScope.$digest();
+
+        expect(classAddSpy.callCount).toBe(2);
+        expect(classRemoveSpy.callCount).toBe(2);
+
+        expect(classAddSpy.calls[0].args[1]).toBe('ng-animate');
+        expect(classAddSpy.calls[1].args[1]).toBe('ng-enter');
+        expect(classRemoveSpy.calls[0].args[1]).toBe('ng-enter');
+        expect(classRemoveSpy.calls[1].args[1]).toBe('ng-animate');
+
+        expect(element.children().length).toBe(100);
+      });
+    });
```

Comment:
I agree that the descriptions needs to be better. Fom what I can see, the new mechanism uses a cache key that combines the animation details and the parent element of the animated element. An invalid animation is one that doesn't have actual animations. $animateCss reads the animation styles at runtime, and if there's no animation or transition duration, there is no animation.
Now if you have 100 elements in a repeater, then all of them have the same parent and (probably) the same classes. If the first element doesn't have a transition / animation, then the next element doesn't have one either, so we can skip getComputedStyle, applying animation classes etc. 

If that's how it's done, then it's a clever way to increase performance when rendering large ngRepeats with ngAnimate, but with no actual animations.
But how would that affect animations if someone applies styles to a specific index in the list? As far as I can see, the animation wouldn't be detected.